evadb/functions/stable_diffusion.py

```diff
@@ -0,0 +1,88 @@
+# coding=utf-8
+# Copyright 2018-2023 EvaDB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import pandas as pd
+
+from evadb.catalog.catalog_type import NdArrayType
+from evadb.functions.abstract.abstract_function import AbstractFunction
+from evadb.functions.decorators.decorators import forward
+from evadb.functions.decorators.io_descriptors.data_types import PandasDataframe
+from evadb.utils.generic_utils import try_to_import_replicate
+
+_VALID_STABLE_DIFFUSION_MODEL = [
+    "sdxl:af1a68a271597604546c09c64aabcd7782c114a63539a4a8d14d1eeda5630c33",
+]
+
+
+class StableDiffusion(AbstractFunction):
+    @property
+    def name(self) -> str:
+        return "StableDiffusion"
+
+    def setup(
+        self,
+        model="sdxl:af1a68a271597604546c09c64aabcd7782c114a63539a4a8d14d1eeda5630c33",
+    ) -> None:
+        assert (
+            model in _VALID_STABLE_DIFFUSION_MODEL
+        ), f"Unsupported Stable Diffusion {model}"
+        self.model = model
+
+    @forward(
+        input_signatures=[
+            PandasDataframe(
+                columns=["prompt"],
+                column_types=[
+                    NdArrayType.STR,
+                ],
+                column_shapes=[(None,)],
+            )
+        ],
+        output_signatures=[
+            PandasDataframe(
+                columns=["response"],
+                column_types=[
+                    NdArrayType.STR,
```

Comment:
The response is a image. So the type of is not `NdArrayType.STR`.

Reply:
The response is the link to an image.

Reply:
I see. Should the response be an image? If the response is the link , how do we compose query? Is the idea we will introduce another generic function `DOWNLOAD_IMAGE` to turn the link into images?

Reply:
I think that might be better. Because I think the links are more efficient to store considering some others are planning to work on semantic caching for this task specifically. I'd like to know your thoughts on this too.

Reply:
Does the link have an expiration time? Or it is always available. If it is permanent, storing the link sounds good!